src/webots/nodes/WbBallJoint.cpp

```diff
@@ -381,6 +381,8 @@ double WbBallJoint::initialPosition(int index) const {
 }
 
 void WbBallJoint::setPosition(double position, int index) {
+  WbHinge2Joint::setPosition(position, index);
+
   if (index == 3) {
```

Comment:
To improve the code readability, I would rather do something like:
```C++
  if (index != 3)
    return;
```
(the same applies for the other types of joints)